x/denommetadata/types/proposal_metadata.go

```diff
@@ -0,0 +1,111 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/x/bank/types"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1beta1"
+)
+
+const (
+	// ProposalTypeCreateDenomMetadata defines the type for a CreateDenomMetadata
+	ProposalTypeCreateDenomMetadata = "CreateDenomMetadata"
+
+	// ProposalTypeUpdateDenomMetadata defines the type for a UpdateDenomMetadata
+	ProposalTypeUpdateDenomMetadata = "UpdateDenomMetadata"
+)
+
+// Assert CreateDenomMetadataProposal implements govtypes.Content at compile-time
+var _ govtypes.Content = &CreateDenomMetadataProposal{}
+var _ govtypes.Content = &UpdateDenomMetadataProposal{}
+
+func init() {
+	govtypes.RegisterProposalType(ProposalTypeCreateDenomMetadata)
+	govtypes.RegisterProposalType(ProposalTypeUpdateDenomMetadata)
+
+}
+
+// NewCreateMetadataProposal creates a new create denommetadata proposal.
+func NewCreateMetadataProposal(title, description string, denommetadata types.Metadata) *CreateDenomMetadataProposal {
+	return &CreateDenomMetadataProposal{
+		Title:         title,
+		Description:   description,
+		TokenMetadata: denommetadata,
+	}
+}
+
+// GetTitle returns the title of a community pool spend proposal.
+func (csp *CreateDenomMetadataProposal) GetTitle() string { return csp.Title }
+
+// GetDescription returns the description of a community pool spend proposal.
+func (csp *CreateDenomMetadataProposal) GetDescription() string { return csp.Description }
+
+// GetDescription returns the routing key of a community pool spend proposal.
+func (csp *CreateDenomMetadataProposal) ProposalRoute() string { return RouterKey }
+
+// ProposalType returns the type of a community pool spend proposal.
+func (csp *CreateDenomMetadataProposal) ProposalType() string { return ProposalTypeCreateDenomMetadata }
+
+// ValidateBasic runs basic stateless validity checks
+func (csp *CreateDenomMetadataProposal) ValidateBasic() error {
```

Comment:
I think the Metadata needs to be validated here as well

Reply:
validation added